bgpd/bgp_attr.c

```diff
@@ -2721,7 +2721,7 @@ static int bgp_attr_encap(struct bgp_attr_parser_args *args)
 		}
 	}
 
-	while (length >= 4) {
+	while (STREAM_READABLE(BGP_INPUT(peer)) >= 4) {
```

Comment:
should length and STREAM_READABLE be the same value?  If so shouldn't that be what we test against?

Reply:
It should be, but in this case (malformed) - it's not the same. Callstack is something like:

```
    #7 0xea536158ccb8 in _zlog_assert_failed /home/ubuntu/frr-public/frr_public_private-libfuzzer/lib/zlog.c:789:2
    #8 0xea53614a3a8c in stream_getc /home/ubuntu/frr-public/frr_public_private-libfuzzer/lib/stream.c:353:3
    #9 0xb67813b31658 in bgp_attr_encap /home/ubuntu/frr-public/frr_public_private-libfuzzer/bgpd/bgp_attr.c:2732:12
```